### PR TITLE
Hide skills list until applicant skills have loaded at least once

### LINKED ISSUE #5392: Bug - Applicant Profile - Skills list indicates no skills during loading state
# Description
While the My Skills page of the Applicant Profile is loading, the list of skills section shows an empty skills message even if the applicant has skills attached to their profile. There should probably be a loading message or some type of loader on the list to indicate that the list is not in its loaded state.

Hide Skill List component if skills are loading AND skills list is empty.

# Steps to Reproduce
1. Go to My Profile
2. Click on My Skills
3. Scroll down to list of skills

# Images
Screen Shot 2021-03-05 at 15 19 21

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #5392.

### Notes:
- Added a `initialRefreshFinished` property to the return value of the web resource hooks. This becomes true when the initial request completes (or starts as true if the initial request is skipped). This should be useful in many similar situations, where we want to show a loading state on first load but not subsequent ones.

### CODE REVIEW

File: resources/assets/js/components/ApplicantProfile/Skills/ProfileSkillsPage.tsx [hunk 1]
```diff
@@ -52,6 +52,12 @@ export const ProfileExperiencePage: React.FC<{ applicantId: number }> = ({
     });
   };
 
+  // The skills list should be only appear after skills have have loaded at least once.
```
Comment (mnigh) [inline resources/assets/js/components/ApplicantProfile/Skills/ProfileSkillsPage.tsx]: Two _haves_ don't make a whole. 😉

File: resources/assets/js/hooks/webResourceHooks/singleResourceHook.ts [hunk 2]
```diff
@@ -108,6 +116,7 @@ export type UseResourceReturnType<T> = {
   value: T;
   status: ResourceStatus;
   error: undefined | Error | FetchError;
+  initialRefreshFinished: boolean; // Becomes true after the intial request is fulfilled or rejected. NOTE: if initial fetch is skipped, this will be set to true immediately.
```
Comment (mnigh) [inline resources/assets/js/hooks/webResourceHooks/singleResourceHook.ts]: Change intial to initial.

**Review by mnigh — CHANGES_REQUESTED** [review]
@tristan-orourke A few trivial changes left: nice work!

**Review by yonikid15 — APPROVED** [review]
lgtm 👍

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**